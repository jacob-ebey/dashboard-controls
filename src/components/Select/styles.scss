@import '../colors.scss';

.input {
  box-sizing: border-box;
  border: 1px solid $color-gray;
  padding: 5px;
  margin: 0px;
  transition: all 0.30s ease-in-out;
  outline: none;
  
  &:focus {
    box-shadow: 0 0 3px $color-primary;
    border: 1px solid $color-primary;
  }
}
