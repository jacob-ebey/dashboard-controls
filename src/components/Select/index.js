import React from 'react';

import styles from './styles.scss';

export default class Component extends React.Component {
  render() {
    const { className, children, ...rest } = this.props;
    return (
      <select
        className={`${styles.input} ${className || ''}`}
        {...rest}
      >
        {children}
      </select>
    );
  }
}
