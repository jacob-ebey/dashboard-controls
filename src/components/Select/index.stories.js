import React from 'react';
import { storiesOf } from '@storybook/react';

import Select from './index';

storiesOf('Select', module)
  .add('default', () => (
    <Select>
      <option value="lol">LOL</option>
      <option value="rofl">ROFL</option>
    </Select>
  ));
