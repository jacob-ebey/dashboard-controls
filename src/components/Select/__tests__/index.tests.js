import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Select from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('components/Select - index.js', () => {
  const createProps = () => ({
    children: <option value="rofl" />,
  });

  test('Renders without error', () => {
    const wrapper = shallow(<Select {...createProps()} />);

    expect(wrapper).toMatchSnapshot();
  });
});