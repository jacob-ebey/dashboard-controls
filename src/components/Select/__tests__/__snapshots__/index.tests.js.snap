// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select - index.js Renders without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component>
    <option
      value="rofl"
    />
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <option
        value="rofl"
      />,
      "className": "input ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "value": "rofl",
      },
      "ref": null,
      "rendered": null,
      "type": "option",
    },
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <option
          value="rofl"
        />,
        "className": "input ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "value": "rofl",
        },
        "ref": null,
        "rendered": null,
        "type": "option",
      },
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
